tests/cli/test_main.py

```diff
@@ -103,13 +106,26 @@ def test_print_output_failure(tmpdir, capsys):
     ppln_run.should_receive('has_succeeded').and_return(False)
     ppln_run.should_receive('status_reason').and_return('failed')
     ppln_run.should_receive('get_error_message').and_return('Build failed ...')
-    (ppln_run
-     .should_receive('get_logs')
-     .and_return([
+
+    log_entries = [
         '2021-11-25 23:17:49,886 platform:- - atomic_reactor.inner - INFO - YOLO 1',
         '2021-11-25 23:17:50,000 platform:- - smth - USER_WARNING - {"message": "user warning"}',
         '2021-11-25 23:17:59,123 platform:- - atomic_reactor.inner - ERROR - YOLO 2',
-     ]))
+     ]
+
+    def get_logs():
+        for log in log_entries:
+            yield log
+        if get_logs_failed:
+            raise Exception("error reading logs")
+
+    ppln_run.should_receive('get_logs').and_return(get_logs())
```

Comment:
How does this test work? `get_logs()` will raise an exception right on this line, how come the test is not failing?

Reply:
Oh nvm, generators